segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/SyncStorage.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage;
+
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import io.pravega.segmentstore.contracts.SegmentProperties;
+import io.pravega.segmentstore.contracts.StreamSegmentException;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.segmentstore.contracts.StreamSegmentSealedException;
+import java.io.InputStream;
+
+/**
+ * Defines an abstraction for Permanent Storage.
+ */
+@SuppressWarnings("checkstyle:JavadocMethod")
```

Comment:
What warning are you getting here when not suppressing? Is it because of the close?

Reply:
All of the methods declare that they `throw StreamSegmentException`, but I wanted the JavaDoc to describe each particular exception in detail (each thrown exception is a sub-class of `StreamSegmentException`). I added the suppress-warnings because checkstyle was complaining that the JavaDoc exception list does not match what the code declares.

I decided to do it this way vs having each method explicitly declare each exception because:
- Some methods may have too many exceptions to declare
- In `RollingStorage`, a single such operation may use several other operations and I'd have to have special error handling for each (i.e., `create` which does not throw `StreamSegmentNotExistsException` needs to call `openWrite` which may throw that exception). Instead of a bunch of intertwined code, I preferred to have the methods declare they throw a common super-exception and have the suppress-warnings.

An alternative would be to have all Segment Store exceptions be runtime exceptions, but that's an issue for another day.